libc/src/math/generic/acoshf16.cpp

```diff
@@ -0,0 +1,88 @@
+//===-- Half-precision acoshf16 function ----------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/math/acoshf16.h"
+#include "src/__support/FPUtil/FEnvImpl.h"
+#include "src/__support/FPUtil/FPBits.h"
+#include "src/__support/FPUtil/cast.h"
+#include "src/__support/FPUtil/except_value_utils.h"
+#include "src/__support/FPUtil/generic/sqrt.h"
```

Comment:
Should probably use this header instead, so that the architecture-specific optimized `sqrt<float>` implementations can be used. You already wrote `libc.src.__support.FPUtil.sqrt` instead of `libc.src.__support.FPUtil.generic.sqrt` in the CMakeLists.txt btw.  

```suggestion
#include "src/__support/FPUtil/sqrt.h"
```

Reply:
Done.